File: pgm/ABC_Java/ABC_350_A.java
import java.io.*;
// import java.math.*;
// import java.util.*;
// import java.util.Map.*;
// import java.util.stream.*;

public class Main {
    public static void main (String[] args) throws Exception {
        BufferedReader reader =
            new BufferedReader (new InputStreamReader (System.in));
        final String S = reader.readLine();
        final String Yes = "Yes";
        final String No = "No";
        final int end = 349;
        final int ng = 316;
        final int zero = 0;
        final int num = Integer.parseInt (S.substring (3, 6));
        if (num <= end && ng != num && zero != num) {
            System.out.println (Yes);
        } else {
            System.out.println (No);
        }
        return;
    }
}

File: pgm/ABC_Kotlin/ABC_350_A.kt
// import java.time.*
// import java.io.BufferedOutputStream
// import java.io.BufferedReader
// import java.io.DataInputStream
// import java.io.FileInputStream
// import java.io.IOException
// import java.io.InputStreamReader
// import java.io.PrintWriter
// import java.util.StringTokenizer

// import kotlin.math.*

const val Yes: String = "Yes"
const val No: String = "No"
const val space: String = " "
const val end: Int = 349
const val cancel: Int = 316
const val zero: Int = 0

fun main() {
    val S: String = readln()
    val edition: Int = S.substring(3, 6).toInt()
    if (edition <= end && edition != cancel && edition != zero) {
        println(Yes)
    } else {
        println(No)
    }
    kotlin.system.exitProcess(0)
}